src/nspawn/nspawn-mount.c

```diff
@@ -438,21 +438,23 @@ static int mount_bind(const char *dest, CustomMount *m) {
                 r = mkdir_parents_label(where, 0755);
                 if (r < 0)
                         return log_error_errno(r, "Failed to make parents of %s: %m", where);
+
+		/* Create the mount point. Any non-directory file can be
+		 * mounted on any non-directory file (regular, fifo, socket,
+		 * char, block).
+		 */
+		if (S_ISDIR(source_st.st_mode))
+			r = mkdir_label(where, 0755);
+		else
+			r = touch(where);
+
+		if (r < 0 && r != -EEXIST)
```

Comment:
the r != -EEXIST check can be dropped now if this is moved into the errno == ENOENT block...

Let's keep this minimal and not keep around code that excepts certain errors if we don't need actually need it.